src/west/cmake.py

```diff
@@ -30,27 +31,35 @@
            'make_c_identifier',
            'CMakeCacheEntry', 'CMakeCache']
 
-DEFAULT_CACHE = 'CMakeCache.txt'
+DEFAULT_CACHE: str = 'CMakeCache.txt'
 
-def run_cmake(args, quiet=False):
+def run_cmake(
+    args: List[str],
+    quiet: bool = False
+) -> None:
     '''Run cmake to (re)generate a build system'''
     cmake = shutil.which('cmake')
     if cmake is None:
         log.die('CMake is not installed or cannot be found; cannot build.')
+        return
     cmd = [cmake] + args
     kwargs = dict()
     if quiet:
         kwargs['stdout'] = subprocess.DEVNULL
         kwargs['stderr'] = subprocess.STDOUT
     log.dbg('Running CMake:', cmd, level=log.VERBOSE_VERY)
     log.dbg('As command:', quote_sh_list(cmd), level=log.VERBOSE_VERY)
-    subprocess.check_call(cmd, **kwargs)
+    subprocess.check_call(cmd, **kwargs)  # type: ignore
```

Comment:
I added the ignore here because `mypy` does not like what is passed to `check_call`. I imagine that when using stuff like args or kwargs it is hard  for the analyzer to guess if arguments are right or not. I think the best would be to implement that part without kwargs, but haven't really gone into more detail.

Reply:
details:
```
src/west/cmake.py:52: error: Argument 2 to "check_call" has incompatible type "**Dict[str, int]"; expected "Union[bytes, str, _PathLike[Any]]"
src/west/cmake.py:52: error: Argument 2 to "check_call" has incompatible type "**Dict[str, int]"; expected "Callable[[], Any]"
src/west/cmake.py:52: error: Argument 2 to "check_call" has incompatible type "**Dict[str, int]"; expected "bool"
src/west/cmake.py:52: error: Argument 2 to "check_call" has incompatible type "**Dict[str, int]"; expected "Union[bytes, str, _PathLike[Any], None]"
src/west/cmake.py:52: error: Argument 2 to "check_call" has incompatible type "**Dict[str, int]"; expected "Union[Mapping[bytes, Union[bytes, str]], Mapping[str, Union[bytes, str]], None]"
```

Reply:
Ok, so this is probably mypy getting a subprocess.check_call type from typeshed which isn't sufficiently general.

Reply:
The function type hints can be found here:
https://github.com/python/typeshed/blob/master/stdlib/3/subprocess.pyi#L454

warning could be solved by changing code to something like:

```python
stdout = subprocess.DEVNULL if quiet else None
sderr = subprocess.STDOUT if quiet else None
subprocess.check_call(cmd, stdout=stdout, stderr=stderr)
```